### ISSUE TITLE
Running with `--features bevy/dynamic` fails silently on WSL2

### ISSUE THREAD

**Issue by cooperuser** [bevyengine/bevy#6723]
## Bevy version

0.9

## \[Optional\] Relevant system information

- Rust version: `cargo 1.67.0-nightly (16b097879 2022-11-14)`
- OS: Windows 11
- WSL: `5.10.102.1-microsoft-standard-WSL2`
- Target: `x86_64-pc-windows-gnu`

## What's happening

I am able to run my project fine without `bevy/dynamic` but when trying to use the feature the program fails silently.

Upon trying to run the .exe manually from the File Explorer I get the following error popup:

![image](https://user-images.githubusercontent.com/21961037/203189104-b99f4ab8-f4dd-4e1a-bbf3-8f37ff8b1717.png)

**Comment by DasLixou** [comment]
I had the same issue on Linux. From my experience I assume that you use a git link for bevy rather than a normal cargo version? For me it only worked with a cargo version

**Comment by cooperuser** [comment]
Nope, just the normal cargo version. I also finally just the `x86_64-pc-windows-msvc` target to work, and the same issue is still happening: runs fine normally, but fails silently with `--features bevy/dynamic`. I see that the `bevy_dylib.dll` exists in the `target/x86_64-pc-windows-msvc/debug` directory, but does not include the base16 string in the name.

**Comment by DasLixou** [comment]
> Nope, just the normal cargo version. I also finally just the `x86_64-pc-windows-msvc` target to work, and the same issue is still happening: runs fine normally, but fails silently with `--features bevy/dynamic`. I see that the `bevy_dylib.dll` exists in the `target/x86_64-pc-windows-msvc/debug` directory, but does not include the base16 string in the name.

I had the same issue when upgrading from 0.8.1 to git. so it seems that its not a problem with git but a commit inbetween 0.8.1 and 0.9. and yes it generates only a `bevy_dylib.dll` but it searches for a dll with iguess a hash at the end. Same behaviour for me on linux so.. yeah.. needs investigation?

**Comment by SonicZentropy** [comment]
Not really a "fix", but if you swap to using Mold as your linker on Linux and get rid of all the dynamic linking in bevy, Mold is about twice as fast as even dynamically linked for iterative recompiles.

**Comment by DasLixou** [comment]
It’s really not a fix. Dylib will you use when you want to ship the gameengine apart from the game so you can update the game without updating bevy without any update

**Comment by cooperuser** [comment]
Does mold work when compiling for either `x86_64-pc-windows-msvc` or `x86_64-pc-windows-gnu`? I haven't been able to get any luck with that (though I haven't tried since getting `lld` to work for the `msvc` target, so I'll try that again when I get the chance)

**Comment by DasLixou** [comment]
> Does mold work when compiling for either `x86_64-pc-windows-msvc` or `x86_64-pc-windows-gnu`? I haven't been able to get any luck with that (though I haven't tried since getting `lld` to work for the `msvc` target, so I'll try that again when I get the chance)

as i know mold has linux support, macos in beta and no windows support yet

**Comment by DasLixou** [comment]
ok so idk why but now it works ._. but im not using mold.
